### ISSUE TITLE
Assigning host to group during runtime

### ISSUE THREAD

**Issue by Wyko** [nornir-automation/nornir#430]
### Question

Is there any way to programmatically add a host to a particular group in order to retrieve the attribute of that group through the host later?

### Use Case

I am trying to make an automated ASA upgrade application. In order to do so, I need to assign specific models a specific upgrade file. My `groups.yaml` looks like this:

```
ASA5515:
  data:
    required_version: 9.8(4)8
    asa_file: asa984-8-smp-k8.bin
    asdm_file: asdm-openjre-7122.bin
    rest_api_file: asa-restapi-132346-lfbff-k8.SPA
```
The group name is the model string returned by a fact getter. That way I can determine programmatically which upgrade file a target device will use rather than hardcoding model numbers for each device. After I get the ASA model number, I then need to assign it to the appropriate group in order to pull the appropriate image files.

**Comment by dbarrosop** [comment]
It is possible but there isn't a direct method to do it. It might be worth adding it though. [InventoryElements](https://github.com/nornir-automation/nornir/blob/develop/nornir/core/inventory.py#L62) (hosts and groups) have a [ParentGroups](https://github.com/nornir-automation/nornir/blob/develop/nornir/core/inventory.py#L46-L54) object. This object contains the list of Groups (by name) the element belongs to and a reference to the object so you will need to add both yourself. However, I'd suggest adding an `add_to_group` method to `InventoryElement` so it's easy to consume.

**Comment by Wyko** [comment]
THanks for the insight! I'm having a hard time understanding the elements of this one though, and I would appreciate a bit of guidance on how to implement that particular function. Otherwise I'll keep plugging away at it :)

**Comment by dbarrosop** [comment]
You need to add a function to `InventoryElements`  similar to:

``` python
def add_to_group(self, name: str, group: "Group") -> None:
    self.parent_groups.add(name, group)
```

and then another one to `ParentGroups` similar to:

``` python
def add(self, name: str, group: "Group") -> None:
    self.refs.append(group)
    self.append(name)
```

It'd be nice to add a `remove` function as well. I haven't tested any of this but I am confident it is quite close to what is needed.

**Comment by jesse-mcgraw** [comment]
Did this idea ever progress any further?  

I find myself needing to add similar functionality and want to avoid re-inventing the wheel if it was already done, or beating a dead horse if it was rejected for some reason

**Comment by dbarrosop** [comment]
no, if you want to open a PR my previous comment details a potential solution

**Comment by halfmetaljacket** [comment]
Suggest closing this issue as resolved by PR #624

**Comment by dbarrosop** [comment]
Good point, thx!
